app/src/lib/repository-matching.ts

```diff
@@ -92,3 +95,47 @@ export function matchExistingRepository(
     }) || null
   )
 }
+
+/**
+ * Check whether or not a GitHub repository matches a given remote.
+ *
+ * The remote's URL might match the cloneURL field of the repository (with a
+ * .git suffix) or it might match the htmlURL field (without a .git suffix)
+ *
+ * @param gitHubRepository the GitHub repository details found from the GitHub API
+ * @param remote the remote details found in the Git repository
+ */
+export function repositoryMatchesRemote(
+  gitHubRepository: GitHubRepository,
+  remote: IRemote
+): boolean {
+  const htmlUrl = gitHubRepository.htmlURL || ''
+
+  if (caseInsensitiveCompare(htmlUrl, remote.url) === 0) {
+    return true
+  }
+
+  const cloneUrl = parseRemote(gitHubRepository.cloneURL || '')
```

Comment:
Same comment here about opting not to parse when null rather than coalescing to an empty string